### PR TITLE
Add My settings /preferences page + testing

### PR DESCRIPTION
**Pull request by laf**
image

### CODE REVIEW

File: resources/views/users/preferences.blade.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                <h3 class="box-title">Device permissions</h3>
+            </div>
+            <div class="box-body">
+                @if (Auth::user()->isAdmin())
+                    <strong class='text-light-blue'>Global Administrative Access</strong>
+                @elseif (Auth::user()->hasGlobalRead())
+                    <strong class='text-green'>Global Viewing Access</strong>
+                @elseif (count($devices))
+                    @foreach ($devices as $device)
+                        <a href="{{ url('/device/'.$device->device_id) }}">{{ $device->hostname }}</a><br />
```
Comment (murrant) [inline resources/views/users/preferences.blade.php]: Don't have to do it right now, but we should set up a helper function or attribute on Device to get the device link.  We can include overlib (or whatever we use) there too.

Comment (laf) [inline resources/views/users/preferences.blade.php]: Yeah just put a standard href in for now as we don't have anything for linking devices yet.

**Comment by murrant** [comment]
:+1: This looks really good. The only thing I thought of, is do we want to list ports the user has access to also?

**Comment by laf** [comment]
I've thought the opposite (ish) in just showing the number of devices / ports rather than a list. As the number of devices grows so will that page and whilst we will paginate it it, is it really worth listing them all on the password change page?

We can link the count of them to /devices and /ports so people can then see what they have access to.

**Comment by murrant** [comment]
@laf: I think that sounds better.

**Comment by laf** [comment]
image

**Comment by laf** [comment]
Done - thanks :)

**Pull request merged**